src/TC/src/NvCodecCliOptions.cpp

```diff
@@ -646,16 +645,17 @@ void NvEncoderClInterface::SetupInitParams(
 
     auto is10bitEncodingSupported = ValidateHighBitDepthEncodingSupport(
         params.encodeGUID, api_func, encoder, descr);
-    
-    // H264 + YUV444_10bit is not a supported use case
-    if (FindAttribute(options, "codec") == "h264" &&
-        FindAttribute(options, "fmt") == "YUV444_10bit" &&
-        FindAttribute(options, "fmt") == "YUV420_10bit") {
+
+    bool is10bit = (FindAttribute(options, "fmt") == "YUV444_10bit") ||
+                   (FindAttribute(options, "fmt") == "YUV420_10bit");
+
+    if(is10bit && !is10bitEncodingSupported){
       throw runtime_error(descr);
     }
-    
-    if (FindAttribute(options, "codec") == "hevc" && !is10bitEncodingSupported) {
-      throw runtime_error(descr); 
+
+    // h.264 doesn't have 10 bit hw enc support;
+    if (FindAttribute(options, "codec") == "h264" && is10bit) {
+      throw runtime_error(descr);
```

Comment:
I like making this check clearer than before but the `descr` of the runtime error is not update with a sensible message right ? Removing the cerr print above was good but the clear message "High Bit Depth Encoding not supported" is no longer thrown.

Reply:
```cpp
    if (FindAttribute(options, "codec") == "hevc" && !is10bitEncodingSupported) {
      throw runtime_error(descr); 
```

I removed this line because `runtime_error` will be thrown if one will encode 8 bit hevc on gpu which doesn't support 10 bit encoding.

Reply:
Yes i get that but the description of the runtime error will not say "h.264 doesn't have 10 bit hw enc support" or something like it. I am only asking for a descriptive error.